esmvalcore/preprocessor/_time.py

```diff
@@ -1292,3 +1298,424 @@ def resample_time(
         raise ValueError(
             f"Time coordinate {dates} does not contain {requested} for {cube}")
     return cube
+
+
+def _lin_pad(array: np.ndarray, delta: float, pad_with: int) -> np.ndarray:
+    """Linearly pad an array on both sides with constant difference."""
+    end_values = (array[0] - pad_with * delta, array[-1] + pad_with * delta)
+    new_array = np.pad(array, pad_with, 'linear_ramp', end_values=end_values)
+    return new_array
+
+
+def _guess_time_bounds(time_coord: DimCoord) -> None:
+    """Guess bounds of time coordinate in-place."""
+    if time_coord.has_bounds():
+        return
+    try:
+        time_coord.guess_bounds()
+    except ValueError:  # coordinate has only 1 point
+        point = time_coord.points[0]
+        time_coord.bounds = [[point - 0.5, point + 0.5]]
+
+
+def _get_lst_offset(lon_coord: Coord) -> np.ndarray:
+    """Get offsets to shift UTC time to local solar time (LST).
+
+    Note
+    ----
+    This function expects longitude in degrees. Can be in [0, 360] or [-180,
+    180] format.
+
+    """
+    # Make sure that longitude is in degrees and shift it to [-180, 180] first
+    # (do NOT overwrite input coordinate)
+    lon_coord = lon_coord.copy()
+    lon_coord.convert_units('degrees')
+    shifted_lon = (lon_coord.points + 180.0) % 360 - 180.0
```

Comment:
Mixed use of decimal (180.0 vs 360) - deliberate? Are the decimals actually needed on this line and the next?

Reply:
I don't think the decimals are really needed, but I think it's generally good practice to use floats instead of integers when floats are desired. Since the longitudes are float64 (as specified by the CMOR tables), this is the case here.

Regarding the use of `360`: a modulo division by a float can behave weirdly (see https://stackoverflow.com/questions/14763722/python-modulo-on-floats), so I use an integer here. But since the numerator is also a float, maybe that doesn't change anything at all. I honestly don't know. I think it's fine to leave it like that.